parse_lib.py

```diff
@@ -95,8 +94,9 @@ def resolve_relative_resource_urls(html_string):
     imgs = html.find_all("img", src=True)
     equations = html.find_all("object", data=True)
     list(map(update_anchor_href, anchors))
```

Comment:
Ok, so my judgement (and @andrewemark and @yujanshrestha please confirm you think the same way) is that maps are rarely if ever used like this---when I saw it, I had to do a double take to understand what was going on.  The list call also makes you think a bit (since its only purpose is to consume the iterable).  After some thought, it is clear that the only possible purpose of the list and map calls is for the side-effects, however I think we are in agreement that just using a loop is clearer.

Definitely terseness is a valuable, but probably not worth the tradeoff.  Also, I think it would be fine to use a one-line for-loop if we cared to:

```
for a in anchors: update_anchor_hrefs(a)
```

Concerning other projects---I don't think I have ever seen a map call used like this before, in Python or Javascript.  That isn't to say it is never done!

I found this SO post about this:

http://stackoverflow.com/questions/14447560/side-effects-in-python-map-python-do-block